src/types.ts

```diff
@@ -5,9 +5,64 @@ import type { SegmentAnalytics } from './extensions/segment-integration'
 export type Property = any
 export type Properties = Record<string, Property>
 
+export const COPY_AUTOCAPTURE_EVENT = '$copy_autocapture'
+
+export const knownUnEditableEvent = [
+    '$web_experiment_applied',
+    '$$client_ingestion_warning',
+    '$feature_enrollment_update',
+    '$feature_flag_called',
+    'survey dismissed',
+    'survey sent',
+    'survey shown',
+    '$snapshot',
+    '$identify',
+    '$groupidentify',
+    '$create_alias',
+] as const
```

Comment:
I don't get it 😅 Why are we being restrictive here? The whole point of this function is to give ultimate flexibility to solve any case that someone could come up with. What if I want to ensure certain properties are filtered out in an identify? What if I want to mask survey sent events to try and remove PII?

I get that these are "important events" but why don't we just document that / provide the list for people to use if they want to. The warning for the $pageview etc feels fine but why not just add these to the warning?

I guarantee someone will have a use case to apply to all events and we will have to add an extra param like "includeUnsafeEvents" or something silly...

Reply:
well, at least some of these are billing related, no?
and $snapshot they definitely can't touch

i'm totally not against moving events between the groups

Reply:
> and $snapshot they definitely can't touch

well,,, $snapshot they mustn't _drop_ and i'd argue shouldn't edit

Reply:
> at least some of these are billing related, no?

🤯 although of course if you sample the survey sent event... then we don't bill you for it, but that's ok because you didn't send it 😅

Reply:
as always you were 99% right 🥇

Reply:
OK, I'm down to only $snapshot events are ignored here

So if you wanted to mask e.g. all URLs. You would have to give us before_send and a mask function in replay

Or we let all events through here and it can be powerful or a foot gun... 🤔